src/Domain/Value/QueryParameter/DateQueryParameter.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of storyblok/php-content-api-client.
+ *
+ * (c) Storyblok GmbH <info@storyblok.com>
+ * in cooperation with SensioLabs Deutschland <info@sensiolabs.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Storyblok\Api\Domain\Value\QueryParameter;
+
+use Webmozart\Assert\Assert;
+
+/**
+ * Represents top level query parameters which are holding simple string values.
+ *
+ * @author Frank Stelzer <dev@frankstelzer.de>
+ */
+abstract readonly class DateQueryParameter extends QueryParameter
```

Comment:
I really think we should not introduce abstract classes here because of the Backwards compatibility promise

Reply:
It makes it way harder to maintain.